src/main/webapp/app/lecture/wizard-mode/lecture-update-wizard.component.html

```diff
@@ -24,38 +21,32 @@
         <div class="flex-grow-1 d-flex justify-content-center">
             <div class="flex-grow-1 d-flex justify-content-space-between ms-5 me-5 code-hint-generation-wrapper">
                 <jhi-lecture-update-wizard-step
-                    [currentlySelected]="isCurrent(1)"
-                    [isPerformed]="isCompleted(1)"
+                    [currentlySelected]="isCurrent(LECTURE_UPDATE_WIZARD_TITLE_STEP)"
+                    [isPerformed]="isCompleted(LECTURE_UPDATE_WIZARD_TITLE_STEP)"
                     [labelTranslationKey]="'artemisApp.lecture.wizardMode.steps.titleStepTitle'"
                 />
                 <div class="connector" [ngClass]="isCompleted(1) ? 'check' : 'unset'"></div>
                 <jhi-lecture-update-wizard-step
-                    [currentlySelected]="isCurrent(2)"
-                    [isPerformed]="isCompleted(2)"
+                    [currentlySelected]="isCurrent(LECTURE_UPDATE_WIZARD_PERIOD_STEP)"
+                    [isPerformed]="isCompleted(LECTURE_UPDATE_WIZARD_PERIOD_STEP)"
                     [labelTranslationKey]="'artemisApp.lecture.wizardMode.steps.periodStepTitle'"
                 />
                 <div class="connector" [ngClass]="isCompleted(2) ? 'check' : 'unset'"></div>
                 <jhi-lecture-update-wizard-step
-                    [currentlySelected]="isCurrent(3)"
-                    [isPerformed]="isCompleted(3)"
+                    [currentlySelected]="isCurrent(LECTURE_UPDATE_WIZARD_ATTACHMENTS_STEP)"
```

Comment:
Does the `isCurrent` method just check the `currentStep`  Attribute? If yes, please replace this function call by a direct conditional. This enables the angular change detection to smartly evaluate the condition only on changes for `currentStep` whereas methods must be called every time. (See client guides for more)

Reply:
I addressed this.

However I could not find any reference to this in the client guidelines.

Reply:
It's the warning in the `0. General` section. (I also experienced that for some reason these big warnings actually get overlooked more easily than the normal text)